Source/DataSources/VelocityVectorProperty.js

```diff
@@ -0,0 +1,173 @@
+        this._subscription = undefined;
+        this._definitionChanged = new Event();
+
+        this.position = position;
+    }
+
+    defineProperties(VelocityVectorProperty.prototype, {
+        /**
+         * Gets a value indicating if this property is constant.
+         * @memberof VelocityVectorProperty.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         */
+        isConstant : {
+            get : function() {
+                return Property.isConstant(this._position);
+            }
+        },
+        /**
+         * Gets the event that is raised whenever the definition of this property changes.
+         * @memberof VelocityVectorProperty.prototype
+         *
+         * @type {Event}
+         * @readonly
+         */
+        definitionChanged : {
+            get : function() {
+                return this._definitionChanged;
+            }
+        },
+        /**
+         * Gets or sets the position property used to compute orientation.
+         * @memberof VelocityVectorProperty.prototype
+         *
+         * @type {Property}
+         */
+        position : {
+            get : function() {
+                return this._position;
+            },
+            set : function(value) {
+                var oldValue = this._position;
+                if (oldValue !== value) {
+                    if (defined(oldValue)) {
+                        this._subscription();
+                    }
+
+                    this._position = value;
+
+                    if (defined(value)) {
+                        this._subscription = value._definitionChanged.addEventListener(function() {
+                            this._definitionChanged.raiseEvent(this);
+                        }, this);
+                    }
+
+                    this._definitionChanged.raiseEvent(this);
+                }
+            }
+        }
+    });
+
+    var position1Scratch = new Cartesian3();
+    var position2Scratch = new Cartesian3();
+    var velocityScratch = new Cartesian3();
+    var timeScratch = new JulianDate();
+    var step = 1.0 / 60.0;
+
+    /**
+     * Gets the value of the property at the provided time.
+     *
+     * @param {JulianDate} [time] The time for which to retrieve the value.
+     * @param {Cartesian3} [result] The object to store the value into, if omitted, a new instance is created and returned.
+     * @returns {Cartesian3} The modified result parameter or a new instance if the result parameter was not supplied.
+     */
+    VelocityVectorProperty.prototype.getValue = function(time, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(time)) {
+            throw new DeveloperError('time is required');
+        }
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new Cartesian3();
+        }
+
+        var property = this._position;
+        if (Property.isConstant(property)) {
+            return undefined;
```

Comment:
The other properties seem to return undefined when there isn't a value at that time. This code was pretty much copied from the `VectorOrientationProperty`. @mramato is returning `undefined` the right thing to do here?